test/generator/implementation_test.rb

```diff
@@ -2,39 +2,54 @@
 
 module Generator
   class ImplementationTest < Minitest::Test
+    # I would love to be mocking these tests, but Minitest::Mock does not play well with
+    # Forwardable. As we'll be refactoring out the includes in Repository soon, doesn't
+    # seem worth the fight.
     FixturePaths = Paths.new(
       metadata: 'test/fixtures/metadata',
       track: 'test/fixtures/xruby'
     )
 
     def test_version
       exercise = Minitest::Mock.new.expect :slug, 'alpha'
-      subject = Implementation.new(paths: FixturePaths, exercise: exercise)
+      repository = Repository.new(paths: FixturePaths, exercise: exercise)
+      subject = Implementation.new(exercise: exercise, repository: repository)
       assert_equal 1, subject.version
     end
 
     def test_update_tests_version
+      # Minitest::Mock doesn't document how to allow the mocked method to be called twice
```

Comment:
You just add another `expect`ation.
```
mock.expect :do_something, 'once'
mock.expect :do_something, 'twice'
```
But you're right about it not being documented :/